### PR TITLE
Add a type for addedBy and fix type issues.

### PR DESCRIPTION
**Pull request by lionell-pack-ttd**

### CODE REVIEW

File: src/web/components/SharingPermission/SharingPermissionsTable.tsx [hunk 1]
```diff
@@ -49,7 +49,9 @@ function DeletePermissionDialog({
   };
 
   const showDeletionNotice = (participant: SharingParticipant) => {
-    const remainSources = participant.addedBy.filter((source) => source !== 'Manually Added');
+    const remainSources = (
+      participant.addedBy.filter((source) => source !== 'Manual') as ParticipantTypeDTO[]
```
Comment (jingyi-gao-ttd) [inline src/web/components/SharingPermission/SharingPermissionsTable.tsx:53]: nit: is there a reason we still keep the string here rather than constant? is that breaking type check?

Comment (lionell-pack-ttd) [reply src/web/components/SharingPermission/SharingPermissionsTable.tsx]: Because we've got type checking now it doesn't really matter - if you get the string wrong it's a type error. I've updated it anyway 🙂

**Review by jingyi-gao-ttd — APPROVED** [review]

File: src/api/entities/Participant.ts [hunk 2]
```diff
@@ -59,7 +59,7 @@ export class Participant extends BaseModel {
 
 // TODO: Can ModelObjectOpt do relationships automatically?
 export type ParticipantDTO = Omit<ModelObjectOpt<Participant>, 'types' | 'users'> & {
-  types?: ModelObjectOpt<ParticipantType>[];
+  types?: ParticipantTypeDTO[];
```
Comment (alex-yau-ttd) [inline src/api/entities/Participant.ts:62]: what's the rationale behind this change?

Comment (lionell-pack-ttd) [reply src/api/entities/Participant.ts:62]: Just because we're already doing the ModelObjectOpt<> to create ParticipantTypeDTO anyway - I think the types would have just "matched", but I'm used to C# where types which are just the same still won't match unless they're the exact same type.
